core/build.gradle

```diff
@@ -45,6 +45,15 @@ test {
     }
 }
 
+ext.moduleName = 'org.bitcoinj.core'
+
+jar {
+    inputs.property("moduleName", moduleName)
```

Comment:
Why not use `instruction "Automatic-Module-Name", "org.bitcoinj.core"` like suggested in the article you linked? What's the difference?

Reply:
`inputs.property("moduleName", moduleName)` tells Gradle to ru-run the `jar` task if `ext.moduleName` changed since the last run.

I'm pretty sure `instruction` in the article is an error (looks like they copied code from a build file using the 'osgi' plugin) -- `attributes` is what is used for plain-old-everyday jars.

Reply:
I see. Also I just saw it's recommended in the Gradle docs: https://guides.gradle.org/building-java-9-modules/#optional_add_code_automatic_module_name_code_manifest_entries_for_all_other_projects

Reply:
FYI @schildbach I got the article corrected: https://disqus.com/home/discussion/branchandbound/automatic_module_name_calling_all_java_library_maintainers_48/#comment-4414097676